api/controllers/ApplicationController.js

```diff
@@ -216,36 +216,41 @@ var applicationController = {
      */
     submitApplication: function(req, res) {
         var id = req.query.merchantReturnData;
-        console.log(id + " - attempting to submit application");
+        sails.log.info(id + " - attempting to submit application");
         Application.findOne({
             where: {
                 application_id: id
             }
         }).then(function(application) {
-            if (application !== null) {
-              console.log(id + " - has returned from barclays");
-              if (application.submitted === 'draft') {
-                console.log(id + " - has not been submitted previously");
-                console.log(id + " - exporting app data");
-                applicationController.exportAppData(req, res);
-              } else {
-                console.log(id + " - has been submitted previously");
-              }
-              if (application.serviceType == 1) {
-                console.log(id + " - displaying standard confirmation page to user");
+            if (application === null) {
+                sails.log.info(id + ' - could not be found in db');
+            }
```

Comment:
```suggestion
            if (application === null) {
                sails.log.info(id + ' - could not be found in db');
                return
            }
```
this block needs to return, otherwise the rest of the code will continue executing (unless you want it to keep executing!)

Reply:
Good spot, didn't know why I didn't pick up on this before 🤦